geoips/plugins/modules/output_checkers/geotiff_checker.py

```diff
@@ -0,0 +1,113 @@
+from geoips.plugins.modules.output_checkers.utils import compare_outputs as co
+
+LOG = logging.getLogger(__name__)
+
+interface = "output_checkers"
+family = "standard"
+name = "geotiff_checker"
+
+
+def correct_type(fname):
+    """Determine if fname is a geotiff file.
+
+    Parameters
+    ----------
+    fname : str
+        Name of file to check.
+
+    Returns
+    -------
+    bool
+        True if it is a geotiff file, False otherwise.
+    """
+    if splitext(fname)[-1] in [".tif"]:
+        return True
+    return False
+
+
+def geotiffs_match(output_product, compare_product):
+    """Use diff system command to compare currently produced image to correct image.
+
+    Parameters
+    ----------
+    output_product : str
+        Full path to current output product
+    compare_product : str
+        Full path to comparison product
+
+    Returns
+    -------
+    bool
+        Return True if images match, False if they differ
+    """
+    # out_diffimg = get_out_diff_fname(compare_product, output_product)
+
+    call_list = ["diff", output_product, compare_product]
+    LOG.info("Running %s", " ".join(call_list))
+    retval = subprocess.call(call_list)
+
+    # subimg_retval = subprocess.call(call_list)
+    if retval != 0:
+        LOG.interactive("    *****************************************")
+        LOG.interactive("    *** BAD geotiffs do NOT match exactly ***")
+        LOG.interactive("    ***   output_product: %s ***", output_product)
+        LOG.interactive("    ***   compare_product: %s ***", compare_product)
+        LOG.interactive("    *****************************************")
+        return False
+
+    LOG.info("    ***************************")
+    LOG.info("    *** GOOD geotiffs match ***")
+    LOG.info("    ***************************")
+    return True
+
+
+def call(compare_path, output_products, test_product_func=None):
+    """Compare the "correct" imagery found the list of current output_products.
+
+    Compares files produced in the current processing run with the list of
+    "correct" files contained in "compare_path".
+
+    Parameters
+    ----------
+    compare_path : str
+        Path to directory of "correct" products - filenames must match output_products
+    output_products : list of str
+        List of strings of current output products,
+        to compare with products in compare_path
+    test_product_func : function, default=None
+        Alternative function to be used for testing output product
+
+          * Call signature must be:
+
+              * output_product, compare_product, goodcomps, badcomps, compare_strings
+
+          * Return must be:
+
+              * goodcomps, badcomps, compare_strings
+
+        * If None, use geoips.compare_outputs.test_product)
+
+    Returns
+    -------
+    int
+        Binary code: 0 if all comparisons were completed successfully.
+    """
+    retval = co.compare_outputs(compare_path, output_products, test_product_func)
```

Comment:
Can this just call "geotiffs_match" directly, without going through the general "compare_outputs" module?  I would assume once we are inside a given output checker, we know what the format is/should be, and can just check it directly?

Reply:
I think as long as this ALWAYS imports compare_outputs from the geoips repo that is probably fine (ie, there is no reference to the explicit output_checker plugins within compare_outputs - it uses get_plugins to implicitly find all the output checker plugins in any repo).  We just don't want to go back to how it was when recenter_tc and geoips repos both had completely separate "compare_outputs" Modules...

Reply:
Actually, maybe the "compare_outputs" utility should be attached to the interface itself, rather than a separate python script in output_checkers/utils ?  @jsolbrig should probably chime in on this.

Reply:
I'll wait to update this and the comment below until we hear back from @jsolbrig. Might be a good idea to attach it to the interface, so I'll wait before I change anything in the 'compare_outputs' code.

Reply:
As an update here, we're discussing how best to handle use of methods from inherited BasePlugin classes. Say we define `OutputCheckersBasePlugin.compare_outputs()`, how can that be accessed from within a plugin's `call()` function.

Currently, a plugin's `call()` function is attached to the plugin's class as a static method like this:
```
obj_attrs["__call__"] = staticmethod(getattr(plugin_module, "call"))
```
This makes is so that the resulting plugin object is callable and will execute `plugin_module.call()` when the object is called. So:
```
plg(foo, bar)
```
calls
```
plg.__call__(foo, bar)
```
which calls
```
plugin_module.call(foo, bar)
```

Since this is implemented as a static method, though, it does not provide access to `self` for the object. This means that we can't access any of the plugin object's instance methods. So, if we define `OutputCheckersBasePlugin.compare_outputs()` we can't access it from within `call()`.

There are a few methods to make instance methods available from within `call()`:

1. We could require all plugin modules' `call()` functions to include one additional argument such as `self` or 
    `plugin`, then attach `call()` to the plugin object as an instance method. using:
    ```
    obj_attrs["__call__"] = instancemethod(getattr(plugin_module, "call"))
    ```
    This would allow us to do something like this:
    ```
    def call(self, foo, bar):
        self.compare_outputs()
    ```
2. Rather than requiring `self` or `plugin` as arguments, we could make them optional, then attach the
    `call()` functions to their classes as staticmethods if `self` or `plugin` is absent and as instancemethods if 
    they are present.
3. We could create an additional method of specifying a plugin by specifying it as a class directly rather than 
    converting a plugin module to a class. So, rather than a plugin module, we could specify:
    ```
    class ImageChecker(OutputCheckersBasePlugin):
        def __call__(self, foo, bar):
            self.compare_outputs()
    ```
4. We could also just stop doing module based plugins and require that all python plugins be written as classes 
    in the first place.

Personally, I prefer option 3 or option 4. 
- Option 3 avoids creating more magic and making module-based plugins more confusing. 
- Option 4 gets rid of all of the magic, but makes implementing a new plugin a little bit more daunting for a non-python programmer and requires lots of changes in lots of places.

I think that option 3 is probably the best compromise solution that avoids the most headaches for everyone including people who have already developed plugins for GeoIPS while option 4 would result in GeoIPS being cleaner and easier to work with.

Reply:
If this just needs an issue opened, let's open the issue, reference it here, then resolve this conversations.

Reply:
This has an issue: https://github.com/NRLMMD-GEOIPS/geoips/issues/362
Resolving.